src/com/timezynk/mongo/util.clj

```diff
@@ -22,12 +22,21 @@
   (reset! mongo-uri {:uri uri
                      :options options}))
 
-(def ^:no-doc make-connection
-  (delay (let [uri (or @mongo-uri
-                       {:uri (System/getenv "MONGO_URI")}
-                       (throw (MongoClientException. "Call set-mongo-uri! or define MONGO_URI environment variable prior to calling wrap-mongo")))]
-           (log/info "MongoDB servers" uri)
-           (connection-method (:uri uri) (:options uri)))))
+(defn current-uri []
+  @mongo-uri)
+
+(def ^:private connection-map
+  (atom {}))
+
+(defn ^:no-doc upsert-connection []
+  (let [uri (or @mongo-uri
+                {:uri (System/getenv "MONGO_URI")}
+                (throw (MongoClientException. "Call set-mongo-uri! or define MONGO_URI environment variable prior to calling wrap-mongo")))]
+    (or (get @connection-map uri)
+        (let [connection (connection-method (:uri uri) (:options uri))]
+          (log/info "MongoDB servers" uri)
```

Comment:
Move `log/info` above the `or` so that it is always printed.

Reply:
Wouldn't that spam the log?

If a client keeps switching between two databases, you would get the message on every swtich. No strong opinion, though. I will move it.